### ISSUE TITLE
Sandbox 0.12.19 refuses a valid key declaration with `EIllegalKeyExpression` in LF 1.4

### ISSUE THREAD

**Issue by jberthold-da** [digital-asset/daml#1326]
I tested with the following simple DAML file `UseKey.daml`
```
daml 1.2
module UseKey where

template UsingAKey with
    op : Party
    id : Text
  where
    key (op, id): (Party, Text) -- sandbox fails with [1]
    maintainer op
    signatory op
```
When loading the resulting `dar` file into sandbox (and triggering a dar load by using `--eager-package-loading`), sandbox throws an exception `EIllegalKeyExpression: ... (AST for that key)`
To reproduce:
```
$ daml install 0.12.19
$ # download sandbox 0.12.19
$ daml damlc package UseKey.daml UseKey
$ java -jar ./sandbox-100.12.19.jar UseKey.dar --eager-package-loading
```

The `dar` load works fine when compiling with 0.12.18 so I assume the problem is related to the LF-1.4 format introduced with 0.12.19. I saw [in the code](https://github.com/digital-asset/daml/blob/master/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala#L206) that LF-1.4 is handled differently when verifying the package. Probably the error is hiding in there?

**Comment by hurryabit** [comment]
https://github.com/digital-asset/daml/pull/1316 fixed the problem. We'll make another SDK release today which contains the fix.

**Comment by hurryabit** [comment]
The release of 0.12.20 can be tracked on https://github.com/digital-asset/daml/pull/1330.

**Comment by cocreature** [comment]
This has been fixed in 0.12.20 which is now released.
